### PR TITLE
JSON value s11n unit tests

### PR DESCRIPTION
**Pull request by tjanc**
Includes minor fixes; not all corner cases are covered by integration tests.

### CODE REVIEW

File: test/refract/test-JsonValue.cc [hunk 1]
```diff
@@ -0,0 +1,1477 @@
...
+        {
+            auto result = generateJsonValue(*el);
+
+            THEN("the result is `42.1930e-39`")
+            {
+                REQUIRE(to_string(result) == "42.1930e-39");
+            }
+        }
+    }
+}
```
Comment (klokane) [inline test/refract/test-JsonValue.cc]: Please add tests for `Nan` and `Inf`

Comment (klokane) [reply test/refract/test-JsonValue.cc]: For NumberElement

**Comment by klokane** [comment]
:+1: great coverage of testcases

File: test/refract/test-JsonValue.cc (hunk 1 above)
Comment (tjanc) [reply test/refract/test-JsonValue.cc]: We use the JSON definition of number, `Nan` and `Inf` are not numbers.

Comment (klokane) [reply test/refract/test-JsonValue.cc]: But you can set value of NumberElement to Nan/Inf, isn't it?
There should be, at least, defined bahavior for this case

Comment (tjanc) [reply test/refract/test-JsonValue.cc]: You can theoretically construct a NumberElement with any string now, as the check for JSON Number is in `Serialize.cc`. But it will have undefined behaviour, so I'm not sure what to test against. Finding something non-numberish in a NumberElement could be handled as an `assert` in `JsonValue.cc`.

Comment (klokane) [reply test/refract/test-JsonValue.cc]: On other side refract is not strictly for JSON, so we should define what happens. e.g. conversion to zero?
Not sure, just idea

Comment (tjanc) [reply test/refract/test-JsonValue.cc]: There is the relevant issue on the MSON spec not being spec enough: https://github.com/apiaryio/mson/issues/93
In API Elements ("refract") we are pretty clear: https://github.com/apiaryio/api-elements/blob/master/docs/element-definitions.md#values
We could add a debug build run time check during JSON serialization (not JSON _value_ serialization), although I don't deem that necessary. Here are the tests currently covering your worries: https://github.com/apiaryio/drafter/blob/master/test/test-Serialize.cc#L931

Comment (klokane) [reply test/refract/test-JsonValue.cc]: Ok. Let it pass for now and create issue to resolve this
